docs/concepts/security.md

```diff
@@ -0,0 +1,60 @@
+# Security
+
+This document explains the security aspects of Kong Ingress Controller.
+
+Kong Ingress Controller communicates with Kubernetes API-server and Kong's
+Admin API. APIs on both sides offer authentication/authorization features
+and the controller integrates with them gracefully.
+
+## Kubernetes RBAC
+
+Kong Ingress Controller is deployed with RBAC permissions as explained in
+[deployment](deployment.md) document.
+It has read and list permissions on most resources but requires update
+and create permission for a few resources to provide seamless integration.
+The permissions can be locked down further if needed depending on the specific
+use-case.
+This RBAC policy is associated with a ServiceAccount in Kubernetes, which
+takes of the Controller authenticating and authorizing against Kuberentes
+API-server.
+
+## Kong Admin API protection
+
+Kong's Admin API is used to control configuration of Kong and proxying behavior.
+If an attacker gets access to Kong's Admin API, Kong will be compromised
+and all bets are off at that point. Hence, it is important that the deployment
+ensures that the likelyhood of this happening is minimized.
+
+In the example deployements, the Controller and Kong's Admin API communicate
+over the loopback(lo) interface of the pod. There is no authorization or
+authentication being done over the loopback listner.
+Although not ideal, this setup is simple to get started and can be further
+hardened.
+
+Please note that it is very important that Kong's Admin API is not accessible
+inside the cluster as any malicious service can change Kong's configuration.
```

Comment:
just to check--did you mean outside instead of inside?

Reply:
Nope, inside is correct here.

What I mean to say:

Deploying Kong Admin API without any sort of authentication and open even inside a cluster or private network is a bad idea. You should always have authentication enabled on the Admin API or not expose it at all.

Reply:
Understood, sounds good.